### ISSUE TITLE
are the testing images used?

### ISSUE THREAD

**Issue by fawazsammani** [sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning#35]
Hi, In your code, you've saved the testing images in a .h5 file. However, you've never used them for evaluation. You've only used the validation images. Do we need to use the testing images for evaluation as well? And Does the paper report results on the testing images?

**Comment by kmario23** [comment]
Ideally, any publishable paper *should* report the scores on a test set (and optionally on the dev set) and I assume that Xu et. al also do the same (the table doesn't explicitly state this). However, it seems they indeed report the test error (after doing K-Fold CV) as can be seen in the lines [L1222](https://github.com/kelvinxu/arctic-captions/blob/master/capgen.py#L1222), [L1334](https://github.com/kelvinxu/arctic-captions/blob/master/capgen.py#L1334), and [L1377](https://github.com/kelvinxu/arctic-captions/blob/master/capgen.py#L1377)

**Comment by fawazsammani** [comment]
@kmario23 Thanks for your reply
I guess @sgrvinod only reported on the validation images to show the concept.

**Comment by sgrvinod** [comment]
Lol, actually, I totally forgot there was a test set. I thought there wasn't. Thanks for bringing this up.

@kmario23 is right that we _should_ report on a test set whenever possible because it's completely uninvolved in the training process and therefore (sorta) reflects real world performance. The validation set is 'tainted' because we optimized – and overfit – the model to it.  

I do know that people are less particular about things like this when it comes to complex deep learning models as opposed to more traditional machine learning applications, which is why you might see validation results reported in some papers. I've no idea if the paper authors used the test set, but they probably did.

At any rate, I quickly ran `eval.py` on `TEST` at beam sizes of 1 and 3, and the results are fairly similar to the validation set – 0.3028 and 0.3306 respectively. I'll update `eval.py` and the Readme tomorrow. If one of you could confirm these numbers, and create a pull request with the changes, that works too.

**Comment by tony-mbw** [comment]
Hi, @sgrvinod @kmario23 According to your answer, the bleu-4 scores are 0.3028 and 0.3306 respectively at beam sizes of 1 and 3, but in the Xu's paper, the bleu-4 score is 0.25 in COCO dataset, Whether the beam search is used in the results of the paper? if so, what is beam size?

**Comment by sgrvinod** [comment]
Hi @tony-mbw, I don't think the results in the paper are with beam search. (This is the same as using beam size of 1.) I think the better score here is mostly because of using the Resnet encoder.

**Comment by guantinglin** [comment]
@tony-mbw 
A poor question...
Why is the value of the bleu score multiplied by a factor of 100 in the paper? 
Is it a general way of representation of bleu scores?

**Comment by kmario23** [comment]
@guantinglin yes, that's correct.
